contrib/bindings/go/pathrs/pathrs.go

```diff
@@ -0,0 +1,239 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package pathrs
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+	"unsafe"
+)
+
+// Open opens directory as a root directory
+func Open(path string) (*Root, error) {
+	r := open(path)
+	err := handleErr(pointerOnRoot, r)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Root{root: r}, nil
+}
+
+// Root is a wrapper on pointer to root api,
+// to use it (safe)
+type Root struct {
+	root cRootPtr
+}
+
+// Resolve resolves name with regard to root path
+func (r *Root) Resolve(path string) (*Handle, error) {
+	handler := resolve(r.root, path)
+	if handler == nil {
+		return nil, handleErr(pointerOnRoot, r.root)
+	}
+	return &Handle{handle: handler}, nil
+}
+
+// Creat creates a file with a such mode by path
+func (r *Root) Creat(path string, mode uint) (*Handle, error) {
+	handler := creat(r.root, path, mode)
+	if handler == nil {
+		return nil, handleErr(pointerOnRoot, r.root)
+	}
+
+	return &Handle{handle: handler}, nil
+}
+
+// Rename renames src to dst, or change flags on this file
+// if the names are the same it's only change the flags
+func (r *Root) Rename(src, dst string, flags int) error {
+	status := rename(r.root, src, dst, flags)
+	if status < 0 {
+		return handleErr(pointerOnRoot, r.root)
+	}
+
+	return nil
+}
+
+// Mkdir creates a directory with a such mode by path
+func (r *Root) Mkdir(path string, mode uint) error {
+	status := mkdir(r.root, path, mode)
+	if status < 0 {
+		return handleErr(pointerOnRoot, r.root)
+	}
+
+	return nil
+}
+
+// Mknod creates a filesystem node named path
+// with attributes mode and dev
+func (r *Root) Mknod(path string, mode uint, dev int) error {
+	status := mknod(r.root, path, mode, dev)
+	if status < 0 {
+		return handleErr(pointerOnRoot, r.root)
+	}
+
+	return nil
+}
+
+// Hardlink creates a hardlink of file named target and place it to path
+func (r *Root) Hardlink(path, target string) error {
+	status := hardlink(r.root, path, target)
+	if status < 0 {
+		return handleErr(pointerOnRoot, r.root)
+	}
+
+	return nil
+}
+
+// Symlink creates a symlink of file named target and place it to path
+func (r *Root) Symlink(path, target string) error {
+	status := symlink(r.root, path, target)
+	if status < 0 {
+		return handleErr(pointerOnRoot, r.root)
+	}
+
+	return nil
+}
+
+// Free frees underling caught resources
+func (r *Root) Free() {
+	/*
+		Should we handle it in this way or always return not nil pointer on root?
+	*/
```

Comment:
Yeah, `Root` should never contain a null pointer.

Reply:
Are you 100% confident about that?
Since this code still returns nil
https://github.com/openSUSE/libpathrs/blob/feceead820ac9252932bff28872db6cfbc092a0b/contrib/bindings/go/pathrs/pathrs.go#L33-L41

Reply:
Sorry, I misread the check -- I thought the check was

```
if r.root != nil {
   free(...)
}
```

Yeah, this is fine -- though can you please fix up the comment so it matches the rest of the coding style?